WordPress/gutenberg
# UnitControl: Fix internal unit parsing to handle incoming unit prop

This update fixes the `<UnitControl />` component to handle (new) incoming `unit` prop changes. This is important for handling interactions such as unit swapping - an interaction we have for the Cover block.

![Screen Capture on 2020-06-26 at 13-19-20 2020-06-26 13_23_23](https://user-images.githubusercontent.com/2322354/85884188-3a644b80-b7b0-11ea-9077-b857d5bbf7a6.gif)

In the GIF demo above, we manually change the unit from `px` to `em`. Afterwards, we **drag to resize** the cover. Dragging automatically switches the unit **back to `px`**.



## How has this been tested?

* Local storybook, Jest, and local Gutenberg

* Run `npm run dev`
* Add a Cover block
* Change the height
* Change the unit to a non `px` value. 
* Drag to resize Cover
* Ensure the unit updates **back** to `px`


## Checklist:
- [n/a] My code follows the accessibility standards. 
- [n/a] I've updated all React Native files affected by any refactorings/renamings in this PR. 

This is a fast followup to https://github.com/WordPress/gutenberg/pull/23006
